### PR TITLE
fix(keeper): avoid docker word false positive in git commits

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary
- Narrow the Docker nested-runtime guard to command-position runtime tokens instead of arbitrary argument text.
- Keep socket marker blocking intact.
- Add regression coverage for git commit messages that mention Docker while still blocking docker after a command separator.

## Live evidence
- 2026-05-06 KST filesystem/runtime retry kmsg_glm-coding-plan_5_1778028299410 resolved repo ACL but stopped at git commit.
- Keeper glm-coding-plan reported git commit failed with sandbox_profile=docker+git_creds blocks nested container runtimes and host socket references.
- The generated commit message included Docker sandbox, which matched the old broad token detector even though no nested container runtime was invoked.
- Storage path is filesystem/JSONL only; PostgreSQL is not part of this evidence path.

## Verification
- ocamlc -stop-after parsing lib/keeper/keeper_shell_docker.ml test/test_keeper_bash_safety.ml
- scripts/dune-local.sh build ./test/test_keeper_bash_safety.exe
- ./_build/default/test/test_keeper_bash_safety.exe
- scripts/check-oas-pin.sh --local-only
- git diff --check

### CODE REVIEW

File: lib/keeper/keeper_shell_docker.ml [hunk 1]
```diff
@@ -145,11 +145,88 @@ let sandbox_socket_markers =
...
+    | Guard_word (word, quoted) :: rest ->
+        if expect_command then
+          if (not quoted) && List.mem word nested_container_runtime_tokens then
+            true
+          else if (not quoted) && (word = "sudo" || word = "command" || word = "time")
+          then
+            scan true false rest
+          else if (not quoted) && word = "env" then
+            scan true true rest
+          else if in_env && (not quoted) && shell_assignment_like word then
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_shell_docker.ml]: 보안 중요 — POSIX shell quote removal 후 docker가 runtime word로 인식됨. fix: Shell_lexer로 tokenize + unquote 후 검사. quoted form bypass 방지.

File: lib/keeper/keeper_shell_docker.ml [hunk 2]
```diff
@@ -145,11 +145,88 @@ let sandbox_socket_markers =
...
+            scan true true rest
+          else
+            scan false false rest
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_shell_docker.ml]: 동의 — env wrapper edge case. fix: skip_env_options helper로 -i / --ignore-environment / -- / VAR=VAL 모두 처리.

File: test/test_keeper_bash_safety.ml [hunk 3]
```diff
@@ -354,6 +355,23 @@ let test_docker_blocks_docker_socket_reference () =
...
+  Alcotest.(check bool)
+    "unquoted docker argument is not a nested runtime unless command-position"
+    false
+    (Keeper_shell_docker.command_uses_nested_container_runtime
+       "git commit -m Docker-sandbox-proof");
+  Alcotest.(check bool)
+    "docker after command separator is still blocked"
+    true
+    (Keeper_shell_docker.command_uses_nested_container_runtime
+       "git status && docker run --rm alpine true")
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply test/test_keeper_bash_safety.ml]: 동의 — bypass + false positive 양쪽 명시 test 추가. quoted runtime msg + quoted runtime cmd 두 case.

File: lib/keeper/keeper_shell_docker.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_shell_docker.ml]: Patched in 97eb0a9b3b: command-position runtime tokens are checked after quote removal, so quoted and partially quoted command words are blocked while commit-message arguments remain allowed.

File: test/test_keeper_bash_safety.ml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeong-sik) [reply test/test_keeper_bash_safety.ml]: Patched in 97eb0a9b3b: added regression assertions for quoted runtime command words, partial quoting, env -i, env --, env -S, inline --split-string, and env value-option false positives.

File: lib/keeper/keeper_shell_docker.ml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_shell_docker.ml]: Patched in 97eb0a9b3b: env scanning now skips env options, assignments, -- terminator, value-taking options, and treats -S/--split-string as a nested command string.

File: lib/keeper/keeper_shell_docker.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_shell_docker.ml]: Fixed in 97eb0a9b3b. The detector now ignores the previous `quoted` bypass and evaluates the normalized command word at command position, so `"docker" run` and `do"cker" run` are blocked while commit-message arguments still pass.

File: lib/keeper/keeper_shell_docker.ml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_shell_docker.ml]: Fixed in 97eb0a9b3b. `env` handling now skips option tokens, `--`, assignment words, value-taking options such as `-u`/`-c`, and also parses `-S`/`--split-string` command strings so `env -i docker ...`, `env -- docker ...`, and split-string docker invocations are blocked.

File: test/test_keeper_bash_safety.ml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeong-sik) [reply test/test_keeper_bash_safety.ml]: Fixed in 97eb0a9b3b. Added regression assertions for quoted command-word docker, partially quoted docker, `env -i`, `env --`, env value-option handling, and `env -S`/`--split-string` cases alongside the original commit-message false-positive tests.

**Review by jeong-sik — COMMENTED** [review]
Taskmaster check: this PR correctly narrows the nested-runtime guard from broad argument-text scanning to command-position token detection, which resolves the false positive where git commit messages mentioning "Docker" triggered the block. The socket marker blocking is preserved. The regression test covering commit-message-mentions-Docker vs. separator-then-docker is the right boundary pair. No structural concerns.
